streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -108,13 +107,20 @@ public void completeRestoration() {
     }
 
     @Override
-    public void prepareSuspend() {
-        log.trace("No-op prepareSuspend with state {}", state());
+    public void suspendDirty() {
+        log.trace("No-op suspend dirty with state {}", state());
+        if (state() == State.RUNNING) {
+            transitionTo(State.SUSPENDED);
+        }
     }
 
     @Override
-    public void suspend() {
-        log.trace("No-op suspend with state {}", state());
+    public void suspendCleanAndPrepareCommit() {
+        log.trace("No-op suspend clean with state {}", state());
+        if (state() == State.RUNNING) {
```

Comment:
Similar here, maybe we could leverage `transitionTo` to help throw the exception.